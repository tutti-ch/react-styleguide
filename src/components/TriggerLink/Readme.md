Whenever we need something that looks like a link, but doesn't take you to a new page, but triggers an action. In these cases it's bad practice to use anchor tags.

Example: open a modal with more details.

### Preview

```jsx noeditor
<TriggerLink>Lick here to open modal</TriggerLink>
```
