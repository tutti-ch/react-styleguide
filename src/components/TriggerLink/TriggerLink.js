import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import classes from "./TriggerLink.scss";

import { filterProps } from "../../helpers/functions";

export default class TriggerLink extends Component {
  static propTypes = {
    /**
     * The content that will be rendered inside the button.
     */
    children: PropTypes.node,

    /**
     * The class name that will be passed to the button.
     */
    className: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.object,
      PropTypes.string
    ])
  };
  render() {
    const { children, className } = this.props;
    const mergedClasses = classNames(className, classes.triggerLink);
    const attrs = filterProps(TriggerLink.propTypes, this.props);

    return (
      <button className={mergedClasses} {...attrs}>
        {children}
      </button>
    );
  }
}
