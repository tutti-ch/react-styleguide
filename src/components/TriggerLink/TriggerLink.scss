@import "~susy/sass/susy";
@import "../../styles/main";

.triggerLink {
  background: none;
  border: 0;
  color: $blue-50;

  &:hover {
    opacity: 0.8;
    text-decoration: underline;
  }
  &:focus {
    outline: none;
  }
}
