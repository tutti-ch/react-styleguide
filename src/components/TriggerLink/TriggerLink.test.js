import React from "react";
import TriggerLink from "./TriggerLink";
import { mount } from "enzyme";

describe("(component) TriggerLink", () => {
  test("snapshot!", () => {
    const component = mount(
      <TriggerLink onClick={() => {}}>Order pizza</TriggerLink>
    );
    expect(component).toMatchSnapshot();
  });
});
