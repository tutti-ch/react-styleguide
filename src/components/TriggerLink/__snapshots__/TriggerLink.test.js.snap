// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(component) TriggerLink snapshot! 1`] = `
<TriggerLink
  onClick={[Function]}
>
  <button
    className=""
    onClick={[Function]}
  >
    Order pizza
  </button>
</TriggerLink>
`;
